### PR TITLE
CSV output updates

### PR DESCRIPTION
**Pull request by isaaccodekill**
### What change is being made?
- Add new functionality for exporting inventory data to CSV by introducing `CSVDownloadService`, updating dependencies to include `csv-stringify`, and refactoring existing types and services to support CSV formatting.

- Include new columns for csv output
```
[
"Inventory Reference",
"GPC Reference Number",
"Subsector name",
"Notation Key",
"Total Emissions",
"Activity Value",
"Activity Units",
"Emission Factor - CO2",
"Emission Factor - CH4",
"Emission Factor - N2O",
"Emission Factor - Unit",
"Datasource ID",
"DataSource name",
]
```

### Why are these changes being made?
These changes enable users to download inventory data in CSV format, a common and widely supported file type that simplifies data manipulation and visualization. The addition of `CSVDownloadService` centralizes the CSV generation logic, keeping the code well-organized and extensible.

> Is this description stale? Ask me to generate a new description by commenting `/korbit-generate-pr-description`

### CODE REVIEW

File: app/src/backend/CSVDownloadService.ts [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+            }
+
+            return {
+              emission_factor_unit: efUnit,
+              emission_co2,
+              emission_ch4,
+              emission_n2o,
+              activity_amount: activityAmount,
+              activity_unit: activityUnit,
+              data_source_id: "N/A",
```
Comment (lemilonkh) [inline app/src/backend/CSVDownloadService.ts:170]: Leaving the field empty in this case might make sense, as if we parse it automatically having N/A strings might complicate things?

**Review by lemilonkh — APPROVED** [review]
Amazing thanks 👍✨

File: app/src/backend/CSVDownloadService.ts (hunk 1 above)
Comment (isaaccodekill) [reply app/src/backend/CSVDownloadService.ts:170]: Good point!

File: app/src/backend/CSVDownloadService.ts [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+                ?.div(new Decimal("1e3"))
+                .toNumber(),
+            };
+          }),
+        };
+      }
+
+      if (dataDictionary[gpcRefNo as string].notation_key) {
+        dataDictionary[gpcRefNo as string].activityValues = [{}];
+      }
```
[the remark this answers is not in the document]
Comment (isaaccodekill) [reply app/src/backend/CSVDownloadService.ts]: in this case, it's not important to care about previously set data

**Pull request merged**